From 734eff2d8b47a144bf0bce7eb7b87aeb8013279f Mon Sep 17 00:00:00 2001
From: Idan Miara <idanm@omnisys.co.il>
Date: Thu, 31 Jan 2019 15:45:05 +0200
Subject: [PATCH] gdal_python.i - adding overviewLevel option to WarpOptions

---
 gdal/swig/include/python/gdal_python.i | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/gdal/swig/include/python/gdal_python.i b/gdal/swig/include/python/gdal_python.i
index 28f41b7825..7882ec1cc4 100644
--- a/gdal/swig/include/python/gdal_python.i
+++ b/gdal/swig/include/python/gdal_python.i
@@ -1220,6 +1220,7 @@ def WarpOptions(options=None, format=None,
          cutlineLayer=None, cutlineWhere=None, cutlineSQL=None, cutlineBlend=None, cropToCutline = False,
          copyMetadata = True, metadataConflictValue=None,
          setColorInterpretation = False,
+         overviewLevel = 'AUTO',
          callback=None, callback_data=None):
     """ Create a WarpOptions() object that can be passed to gdal.Warp()
         Keyword arguments are :
@@ -1259,6 +1260,7 @@ def WarpOptions(options=None, format=None,
           copyMetadata --- whether to copy source metadata
           metadataConflictValue --- metadata data conflict value
           setColorInterpretation --- whether to force color interpretation of input bands to output bands
+          overviewLevel --- To specify which overview level of source files must be used
           callback --- callback method
           callback_data --- user data for callback
     """
@@ -1356,6 +1358,19 @@ def WarpOptions(options=None, format=None,
             new_options += ['-cvmd', str(metadataConflictValue)]
         if setColorInterpretation:
             new_options += ['-setci']
+        if overviewLevel is not None:
+            if _is_str_or_unicode(overviewLevel):
+                overviewLevelStr = overviewLevel
+            else:
+                overviewLevelStr = ''
+                if overviewLevel is None:
+                    overviewLevelStr = 'NONE'
+                if isinstance(overviewLevel, int):
+                    if overviewLevel < 0:
+                        overviewLevelStr = 'AUTO'
+                    overviewLevelStr = overviewLevelStr + str(overviewLevel)
+            if overviewLevelStr != '':
+                new_options += ['-ovr', overviewLevelStr]
 
     return (GDALWarpAppOptions(new_options), callback, callback_data)
 
-- 
2.16.1.windows.4

